src/main/java/com/codeless/api/automation/mapper/TestDtoToTestMapper.java

```diff
@@ -30,4 +34,8 @@ private String toString(List<Map<Object, Object>> json) {
       throw new RuntimeException();
     }
   }
+
+  private String getUserName() {
```

Comment:
I would prefer to propagate username from controller to where ever it is needed so that it is obvious from where user is taken.

Reply:
Do not we use schedules I do not see changes over there`?

Reply:
Updated to following solution:
`public ResponseEntity<Test> createTest(@RequestBody @Valid Test test, Principal principal)`
works like a charm